pkg/serverless/executioncontext/executioncontext.go

```diff
@@ -72,13 +78,34 @@ func (ec *ExecutionContext) SetFromInvocation(arn string, requestID string) {
 	ec.arn = strings.ToLower(arn)
 	ec.lastRequestID = requestID
 	if len(ec.coldstartRequestID) == 0 {
-		ec.coldstart = true
+		// Putting this within the guard clause of
+		// coldstartRequestID == 0 ensures this is set once
+		// 10 seconds is the maximimum init time, so if the difference
+		// between the first invocation start time and initTime
+		// is greater than 10, we're in a proactive_initialization
+		// not a cold start
```

Comment:
Question: is there a chance that a proactive init can happen within less than 10s if the initialization is very very fast? In that case, what does the trace look like?